share/spack/gitlab/cloud_pipelines/.gitlab-ci.yml

```diff
@@ -69,6 +69,9 @@ default:
 .base-job:
   variables:
     SPACK_BUILDCACHE_DESTINATION: "s3://spack-binaries/${CI_COMMIT_REF_NAME}/${SPACK_CI_STACK_NAME}"
+  id_tokens:
+      GITLAB_OIDC_TOKEN:
+        aud: "${OIDC_TOKEN_AUDIENCE}"
```

Comment:
@scottwittenburg is this legal? I can't use `id_tokens` in the `if` blocks below, so instead I set it up top and reference a variable I set in the if blocks.

Reply:
Are you asking about whether variables are expanded in `aud` there?  Or if the `id_tokens` is evaluated before/after the `OIDC_TOKEN_AUDIENCE` variable? I'm not familiar with the `id_tokens` stuff, but I can check the docs.  

If that doesn't prove conclusive, can we test this stuff in staging or a side project in production to get a definitive answer?

Reply:
> Are you asking about whether variables are expanded in aud there? Or if the id_tokens is evaluated before/after the OIDC_TOKEN_AUDIENCE variable?

I was asking about the latter, but maybe I took the former for granted. I'll test it on a staging job.

Reply:
I checked the [docs](https://docs.gitlab.com/ee/ci/yaml/#id_tokens), they say you can use ci/cd variables in the `aud` claim.  

Regarding the order of evaluation: we override variables set up top in the specific rules, which makes me suspect what you have here might not work.  But if you can test in staging, that's obviously best.

Reply:
```
  id_tokens:
    GITLAB_OIDC_TOKEN:
      aud: "${FOO}"

  rules:
    - if: $CI_COMMIT_BRANCH =~ /^bar/
      when: always
      variables:
        FOO: "bar"
    - if: $CI_COMMIT_BRANCH =~ /^baz/
      when: always
      variables:
        FOO: "baz"
```

This worked for me on staging 😮.